torch/fx/experimental/proxy_tensor.py

```diff
@@ -527,3 +531,51 @@ def wrap_fake_symbolic(x, sym_shape):
         return t
 
     return wrapped
+
+
+maybe_disable_proxy_tensor_mode = functools.partial(maybe_disable_tensor_mode, ProxyTorchDispatchMode)
+
+
+def get_isolated_graphmodule(func, args, kwargs):
+    """A helper function used to get the GraphModule for the given func.
+
+    It's expected to be used in the ProxyTensor tracing context.
+    It detaches the args and kwargs from the current tracer so that the trace of
+    the current graph module can be created without any side-effects.
+    """
+    # make_fx doesn't support kwargs, so we need to do this flattening
+    # and then unflatten the args before calling func
+    all_args, spec = pytree.tree_flatten((args, kwargs))
+
+    def wrapped(args):
+        fn_args, fn_kwargs = pytree.tree_unflatten(args, spec)
+        return func(*fn_args, **fn_kwargs)
+
+    # create a new tracer object
+    graph = torch.fx.Graph()
+    new_tracer = PythonKeyTracer()
+    new_tracer.graph = graph
+
+    # detach proxy tensors from the current tracer
+    detached_all_args = [
+        ProxyTensor(
+            a.elem,
+            torch.fx.Proxy(
+                graph.create_node(
+                    a.proxy.node.op,
+                    a.proxy.node.target,
+                    a.proxy.node.args,
+                    a.proxy.node.kwargs,
+                    a.proxy.node.name,
+                ),
+                tracer=new_tracer,
+            ),
+        )
+        if isinstance(a, ProxyTensor) else a for a in all_args
+    ]
```

Comment:
Wasn't the plan here to add a flag to the proxy tensor mode to disable tracing, so that you wouldn't have to all of this detaching business? If I understand correctly, you're going to end up with a malformed FX graph becausethese nodes refer to nodes that aren't actually in the graph.

Reply:
A new graph is being created here and new "placeholder" nodes are created on the graph with `graph.create_node(...)`, so the nodes are actually in the graph, maybe it would be less confusing if I wrote `graph.placeholder(a.proxy.node.target)`. The resulting graph is not malformed.

For normal (not tensor factory functions) the trace is recorded not with the `ProxyTorchDispatchMode`, but with `torch.fx.Proxy`'s tracer and its `__torch_function__` override. Therefore we need new Proxy objects here, detached from the outer tracer in `ProxyTensor.proxy.tracer`. That's an explicit approach. Now with disabled ambient ProxyTorchDispatchModes a new isolated tracer and proxies are created implicitly inside the `make_fx` call just with unwrapped `ProxyTensor.elem`, so I will change the code to use that.